---
 src/drmgr/drmgr.c |   56 +++++++++++++++++++++++-------------------------------
 1 file changed, 24 insertions(+), 32 deletions(-)

Index: ppu_cleanup/src/drmgr/drmgr.c
===================================================================
--- ppu_cleanup.orig/src/drmgr/drmgr.c	2014-01-31 14:25:49.000000000 -0600
+++ ppu_cleanup/src/drmgr/drmgr.c	2014-02-03 13:44:05.000000000 -0600
@@ -299,54 +299,45 @@
 	return -1;
 }
 
-struct command *
-parse_and_validate_options(int argc, char *argv[], struct options *opts)
+int main(int argc, char *argv[])
 {
-	struct command *command = NULL;
-	int rc;
+	struct options opts;
+	char log_msg[DR_PATH_MAX];
+	struct command *command;
+	int i, rc, offset;
+
+	rc = dr_init();
+	if (rc)
+		return rc;
 
-	parse_options(argc, argv, opts);
+	parse_options(argc, argv, &opts);
 
 	if (display_capabilities) {
 		print_dlpar_capabilities();
-		return NULL;
+		dr_fini();
+		return 0;
 	}
 
-	command = get_command(opts);
+	command = get_command(&opts);
 
 	if (display_usage) {
 		command_usage(command);
-		return NULL;
+		dr_fini();
+		return 0;
 	}
 
 	/* Validate the options for the action we want to perform */
-	rc = command->validate_options(opts);
-	if (rc)
-		return NULL;
+	rc = command->validate_options(&opts);
+	if (rc) {
+		dr_fini();
+		return -1;
+	}
 
 	/* Validate this platform */
-	if (!valid_platform("chrp"))
-		return NULL;
-
-	return command;
-}
-
-int
-main(int argc, char *argv[])
-{
-	struct options opts;
-	char log_msg[DR_PATH_MAX];
-	struct command *command;
-	int i, rc, offset;
-
-	rc = dr_init();
-	if (rc)
-		return rc;
-
-	command = parse_and_validate_options(argc, argv, &opts);
-	if (!command) {
+	rc = valid_platform("chrp");
+	if (rc) {
 		dr_fini();
-		return -1;
+		return rc;
 	}
 
 	set_timeout(opts.timeout);
@@ -363,5 +354,6 @@
 	/* Now, using the actual command, call out to the proper handler */
 	rc = command->func(&opts);
 
+	dr_fini();
 	return rc;
 }
